core/provisioning-java/src/main/java/org/apache/syncope/core/provisioning/java/pushpull/AbstractPushResultHandler.java

```diff
@@ -179,6 +181,10 @@ protected void link(final Any<?> any, final boolean unlink, final ProvisioningRe
                 operation(unlink ? PatchOperation.DELETE : PatchOperation.ADD_REPLACE).
                 value(profile.getTask().getResource().getKey()).build());
 
+        if (any instanceof Group && ((Group) any).getUDynMembership() != null) {
```

Comment:
You should do the same for `adynMembershipConds`

Moreover, it could make sense to centralize the changes you are introducing in a single method like as

```java
protected void copyDynMembershipConds(final Any<?> any, final AnyUR req) {
if (any instanceof Group && ((Group) any).getUDynMembership() != null) {
            ((GroupUR) req).setUDynMembershipCond(((Group) any).getUDynMembership().getFIQLCond());

            // also copy aDynMembershipConds
        }
}
```